libraries/Update/src/HttpsOTAUpdate.cpp

```diff
@@ -0,0 +1,85 @@
+/* Cloud task
+
+   This example code is in the Public Domain (or CC0 licensed, at your option.)
+
+   Unless required by applicable law or agreed to in writing, this
+   software is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+   CONDITIONS OF ANY KIND, either express or implied.
+*/
+#include <stdio.h>
+#include <string.h>
+
+#include <freertos/FreeRTOS.h>
+#include <freertos/task.h>
+
+#include "esp32-hal-log.h"
+#include "esp_http_client.h"
+#include "esp_https_ota.h"
+
+#include"HttpsOTAUpdate.h"
+#include"Esp.h"
+
+static TaskHandle_t ota_task_handle = NULL;
+const char * server_cert;
+const char * ota_url;
+bool skip_cert_common_name_check;
```

Comment:
Can all the above variables be static?

Reply:
Or, instead can you directly pass them (through a structure or something) through `xTaskCreate` in `start_firmware_upgrade`